test/buy-tshirt.spec.ts

```diff
@@ -1,41 +1,66 @@
+  AddressStepPage,
+  ShippingStepPage,
+  PaymentStepPage,
+  BankPaymentPage,
+  OrderResultPage
+} from '../src/page';
 
 describe('Buy a t-shirt', () => {
+  const emailKey: string = 'aperdomobo@gmail.com';
+  const passwordKey: string = 'WorkshopProtractor';
+  const menuContentPage: MenuContentPage = new MenuContentPage();
+  const productListPage: ProductListPage = new ProductListPage();
+  const productAddedModalPage: ProductAddedModalPage = new ProductAddedModalPage();
+  const orderSummary: OrderSummaryPage = new OrderSummaryPage();
+  const signInCardStep: SignInCardStepPage = new SignInCardStepPage();
+  const addressStepPage: AddressStepPage = new AddressStepPage();
+  const shippingStepPage: ShippingStepPage = new ShippingStepPage();
+  const paymentStepPage: PaymentStepPage = new PaymentStepPage();
+  const bankPaymentPage: BankPaymentPage = new BankPaymentPage();
+  const orderResultPage: OrderResultPage = new OrderResultPage();
+
   beforeEach(() => {
     jasmine.DEFAULT_TIMEOUT_INTERVAL = 120000;
   });
 
   it('then should be bought a t-shirt', async () => {
     await browser.get('http://automationpractice.com/');
     await (browser.sleep(10000));
-    await $('#block_top_menu > ul > li:nth-child(3) > a').click();
-    await (browser.sleep(3000));
-    await $('#center_column a.button.ajax_add_to_cart_button.btn.btn-default').click();
+    await menuContentPage.goToTshirtMenu();
     await (browser.sleep(3000));
-    await $('[style*="display: block;"] .button-container > a').click();
+    await productListPage.addTshirtToCart();
     await (browser.sleep(3000));
-    await $('.cart_navigation span').click();
+    await productAddedModalPage.goToShoppingCartSummary();
     await (browser.sleep(3000));
 
-    await $('#email').sendKeys('aperdomobo@gmail.com');
-    await $('#passwd').sendKeys('WorkshopProtractor');
-    await $('#SubmitLogin > span').click();
+    await orderSummary.goToLoginStep();
+    await (browser.sleep(3000));
+    await signInCardStep.sendEmailandPasswKeys(emailKey, passwordKey);
+    await signInCardStep.signIn();
     await (browser.sleep(3000));
 
-    await $('#center_column > form > p > button > span').click();
+    await addressStepPage.goToShippingStep();
     await (browser.sleep(3000));
 
-    await $('#cgv').click();
+    await shippingStepPage.acceptTermsOfService();
     await (browser.sleep(3000));
 
-    await $('#form > p > button > span').click();
+    await shippingStepPage.goToPaymentStepPage();
     await (browser.sleep(3000));
-    await $('#HOOK_PAYMENT > div:nth-child(1) > div > p > a').click();
-    await $('#cart_navigation > button > span').click();
+    await paymentStepPage.payByCheck();
     await (browser.sleep(3000));
-
+    await bankPaymentPage.confirmOrder();
     await (browser.sleep(3000));
-    await expect($('#center_column > div > p > strong').getText()).toBe('Your order on My Store is complete.');
+
+    await expect(await orderResultPage.getOrderResult())
```

Comment:
I think that the second await is too much. It is enough with the await inside the orderResultPage.getOrderResult method

Reply:
You're right @leonleo997. Thanks for suggestion